src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/reverse_shape_and_type_infer.hpp"
+
+#include "itt.hpp"
+#include "openvino/opsets/opset10.hpp"
+
+using namespace ov::opset10;
+
+namespace {
+bool inherit_output_shape(std::shared_ptr<ov::Node> node, std::vector<size_t> input_idxs) {
+    auto is_changed = false;
+    auto output_shape = node->get_output_partial_shape(0);
+
+    for (auto idx : input_idxs) {
+        if (node->get_input_partial_shape(idx).rank().is_dynamic()) {
+            OPENVINO_SUPPRESS_DEPRECATED_START
+            node->get_input_tensor(idx).set_partial_shape(output_shape);
+            OPENVINO_SUPPRESS_DEPRECATED_END
+            is_changed = true;
+        }
+    }
+    return is_changed;
+}
+
+bool inherit_output_rank(std::shared_ptr<ov::Node> node, std::vector<size_t> input_idxs) {
+    auto is_changed = false;
+    auto output_shape = node->get_output_partial_shape(0);
+
+    for (auto idx : input_idxs) {
+        if (idx < node->get_input_size() && node->get_input_partial_shape(idx).rank().is_dynamic()) {
+            OPENVINO_SUPPRESS_DEPRECATED_START
+            node->get_input_tensor(idx).set_partial_shape(ov::PartialShape::dynamic(output_shape.rank()));
+            OPENVINO_SUPPRESS_DEPRECATED_END
+            is_changed = true;
+        }
+    }
+    return is_changed;
+}
+
+bool inherit_output_type(std::shared_ptr<ov::Node> node, std::vector<size_t> input_idxs) {
+    auto is_changed = false;
+    auto output_type = node->get_output_element_type(0);
+
+    for (auto idx : input_idxs) {
+        if (node->get_input_element_type(idx).is_dynamic()) {
+            OPENVINO_SUPPRESS_DEPRECATED_START
+            node->get_input_tensor(idx).set_element_type(output_type);
```

Comment:
Please add convert or change element type of parameter op

Reply:
To do that we need to propagate the type and shape up. That will eventually happen on lines 69-77